tools/stage_time_analysis.groovy

```diff
@@ -0,0 +1,82 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import groovy.json.JsonSlurper
+
+  def getJobs(String stats_file, String find_stage, String folder, Object list_jobs) {
+    list_jobs.each { job ->
+      if (job.jobs && job.jobs.size() > 0) {
+        if (folder.equals("")) {
+          getJobs(stats_file, find_stage, "job/${job.name}", job.jobs)
+        } else {
+          getJobs(stats_file, find_stage, "${folder}/job/${job.name}", job.jobs)
+        }
+      } else {
+        def jobName="${folder}/job/${job.name}"
+        if (folder.equals("")) {
+          jobName="job/${job.name}"
+        }
+        println("${jobName}")
+
+        def workflow = null
```

Comment:
extract this section to its own function?

Reply:
extracted to a fn